User Login Credentials

jameslopez815 opened the pull request:
Username: james
Password: launch

Admin Login Credentials
Username: admin
Password: code

jameslopez815 committed:
User Login Credentials
Username: james
Password: launch

Admin Login Credentials
Username: admin
Password: code

jameslopez815 committed:
Rename imdex.html to home.html

werfelmann reviewed (changes requested):
Thanks, James.

Great work on this! I noticed a two things we should discuss before merging this.

1. I created a User entity when I set up the database. I see you've added a MyUser entity.  I don't think we need both of these. I included those fields (id, username, password, and role, though I implemented the role as an enum. I also added some validation constraints. Could you please take a look at that and perhaps we can consolidate?

2. I also have a UserRepository -- It's very similar to yours, but key differences are I extended the CrudRepository interface rather than the JpaRespository interface, and I did not include the field for an Optional<> object. Again, I think we should consolidate these. Really not sure what the differences are on the JpaRepository and the CrudRepository.

Thanks so much for pushing this. The Java Security stuff looks pretty interesting.

DW

jaamesh commented:
Great work, James!  Two questions:

1. If we include org.springframework.boot:spring-boot-starter-security, I think it prevents us from accessing the Spring Boot application from the React application.  Did you figure out a way to configure spring-boot-starter-security so that we can connect the two apps?
2. What is the plan to integrate the Spring Boot pages like custom_login.html with the React application?

jameslopez815 commented:
David and I decided not to merge the current User Login Credentials Pull Request. We decided it would be best to refactor the MyUser class into the User class. I completed this task this morning and also included user registration. However, the issue I’m now facing is that the registered users are not appearing in the MySQL database. I’ll continue to work on said issue this evening. 

As for your questions:
I removed 'org.springframework.boot:spring-boot-starter-security' and replaced it with 'org.springframework.security:spring-security-crypto'. The change should resolve any issues. 

All custom_login.html pages will live in a static directory. We have all the necessary Spring Boot dependencies to render HTML, including Spring Web and Thymeleaf.

jaamesh committed:
resolving merge conflict
